.github/workflows/cd.yaml

```diff
@@ -0,0 +1,119 @@
+name: cd
+
+on:
+  push:
+    tags:
+      - "v[0-9]+.[0-9]+.[0-9]+"
+  workflow_run:
+    workflows:
+      - ci
+    types:
+      - completed
+    branches:
+      - master
+
+env:
+  TERM: xterm
+  GO_VERSION: 1.19.6
+
+jobs:
+  build:
+    runs-on: ubuntu-22.04
+    needs:
+      - lint
+    steps:
+      - name: ci/checkout-repo
+        uses: actions/checkout@ac593985615ec2ede58e132d2e21d2b1cbd6127c # v3.3.0
+
+      - name: ci/setup-go
+        uses: actions/setup-go@6edd4406fa81c3da01a34fa6f6343087c207a568 # v3.5.0
+        with:
+          go-version: "${{ env.GO_VERSION }}"
+          cache: true
+
+      - name: ci/ensure-build-on-all-platforms
+        run: make dist
+
+      - name: ci/generate-release-notes
```

Comment:
I have seen this multiple times in different repos. Let's mark this to be an action or even do it right now.